MessageQueues/MessageQueues.ProducerApp/FileWatcherService.cs

```diff
@@ -0,0 +1,74 @@
+﻿using MessageQueues.Models;
+using RabbitMQ.Client;
+using Constants = MessageQueues.Models.Constants;
+
+namespace MessageQueues.ProducerApp;
+
+public class FileWatcherService : IDisposable
+{
+    private const int ChunkSize = 2 * 1024 * 1024;
+    private readonly IModel _model;
+    private readonly FileSystemWatcher _watcher;
+
+    public FileWatcherService(IModel model, string path, string filter)
+    {
+        if (!Directory.Exists(path))
+            throw new DirectoryNotFoundException("Path doesn't exist");
+
+        _model = model;
+
+        _watcher = new FileSystemWatcher(path, filter);
+        _watcher.EnableRaisingEvents = true;
+
+        _watcher.Created += OnCreated;
+    }
+
+    private void OnCreated(object sender, FileSystemEventArgs e)
+    {
+        Console.WriteLine("Processing '{0}'", e.FullPath);
+
+        var fileSize = new FileInfo(e.FullPath).Length;
+        var chunksCount = (int)Math.Ceiling((double)fileSize / ChunkSize);
+
+        Console.WriteLine("Chunks count - {0}", chunksCount);
+
+        using var fs = new FileStream(e.FullPath, FileMode.Open, FileAccess.Read);
```

Comment:
Just check according this article (I've faced this issue on my mentoring program). Large files are in use several milliseconds after event was rised.
https://stackoverflow.com/questions/61646218/filesystemwatcher-c-sharp-cannot-access-file-because-it-is-being-used-by-anoth